docs/core-concepts.md

```diff
@@ -96,18 +195,119 @@ They typically record any external side effects they've triggered as events in t
+        │             │       ┌───────▶│  Projector  │             │             │
+     ┌─▶│  Aggregate  │   │   │        │             │    │        │Query Handler│
+     │  │             │       │        └─────────────┘             │             │
+     │  └─────────────┘   │  D. Read          │           │        └─────────────┘
+     │         │              event      E. Update                        ▲
+ A. Load   C. Emit        │   │          Projection       │               │
+state from  Event             │               │                       G. Read
+  events       │          │   │               │           │          Projection
+     │         ▼              │               ▼                           │
+     │  ┌─────────────┐   │   │        ┌─────────────┐    │               │
+     │  │             │       │        │             │                    │
+     └──│ Event Store │───┼───┘        │ Projection  │────┼───────────────┘
+        │             │                │             │
+        └─────────────┘   │            └─────────────┘    │
+
+                          │                               │
 
-![Process View](./images/process-view.png)
+```
+
+### 1. Handling a Command
 
-Note that clients may instead choose to run each ESP in their own process directly. The coordination of this is not currently provided by EventSourcery. 
+A command comes into the application and is routed to a command handler. The command handler initialises an aggregate and loads up its state from events in the store. The command handler then defers to the aggregate to handle the command. It then stores any new events raised by the aggregate into the event store.
+
+```ruby
+class AddTodoCommandHandler
+  def handle(id:, title:, description:)
+    # The repository provides access to the event store for saving and loading aggregates
+    repository = EventSourcery::Repository.new(
+      event_source: EventSourcery.config.event_source,
+      event_sink: EventSourcery.config.event_sink,
+    )
+
+    # Load up the aggregate from events in the store
+    aggregate = repository.load(TodoAggregate, id)
+
+    # Defer to the aggregate to execute the add command.
+    # This may raise new events in the aggregate which we'll need to save.
+    aggregate.add(title, description)
+
+    # Save any newly raised events back into the event store
+    repository.save(aggregate)
+  end
+end
+```
 
-## Query Handlers
+### 2. Updating a Projection
 
-The web layer will take a query and pass it to the appropriate query handler. The query handler reads from a projection into a model and returns it as JSON over the web.
+Projecting is process of converting (or collecting) a stream of events into a structural representation. You can think of the process as a fold over a sequence of events. You can think of a projection as a read model that is generally persisted somewhere like a database table.
```

Comment:
It would be hard for me to understand what converting "a stream of events into a structural representation" means :)